documentation/concepts/determinism/index.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: sv_wiki
+title: Modular determinism 
+---
+
+## Running the analysis
+
+With a file such as:
+
+```
+grammar determinism
+
+import host:grammar;
+
+copper_mda testGrammar(hostParser) {
+   extension:grammmar;
+}
+
+copper_mda testGrammarTwo(hostParser) {
+   extension:grammmar_two;
+}
+
+... etc 
+```
+
+Run ```silver --clean determinism```
+
+Unlike in well-definedness, determinism tends to report errors quickly. Determinism will usually stop evaluating errors after the first error it reports, as a lot of errors are reported as bugs or crashes in copper (ideally this will change). Determinism will also not evaluate any grammars following one that fails in a list of ```copper_mda``` definitions, and will evaluate those definitions in order from top to bottom.
+
+---
+
+### Types of Errors Reported
+
+1. <... a long stack trace> ... Requested head of nil.
+
+   This is a bug in copper, where it doesn't check the length of a list before obtaining the first element of the list.
+
+   ##### Potential solutions:
+
+   Consider changing 'imports' to 'exports' on terminals from other grammars.
+
+2. Missing Bridge Productions 
+
+   Example:
+   ```
+   ... Invalid content was found starting with element 'ProductionRef'. No child element is expected at this point.
+   ... [grammar ext]: Extension grammar is missing the following required attributes: [bridgeProductions]
+   ```  
+
+   This indicates that the automatic scan copper performs to find bridge productions, which are productions that begin with marking terminals, was unsuccessful. Either:  
```

Comment:
Silver performs, actually.

And maybe instead of "unsuccessful", say that it didn't find any bridge productions.